addons/html_editor/static/src/main/list/utils.js

```diff
@@ -28,6 +28,9 @@ export function insertListAfter(document, afterNode, mode, content = []) {
  * - container for nested lists (li.oe-nested)
  */
 export function compareListTypes(a, b) {
+    if (!a || !b) {
```

Comment:
Isn't that weird that we'd call `compareListTypes` with a falsy value? Is it something new due to your changes or was it already the case before?

Reply:
Steps to reproduce:

1. Set up the following HTML:
   ```html
   <ul>
     <li class="oe-nested">abcd</li>
     <li class="oe-nested o-we-hint" placeholder="List">[]<br></li>
   </ul>
   ```
2. Press the Tab key.
3. This results in a traceback: "Cannot read the tagName of null."

Video reference: [View here](https://www.awesomescreenshot.com/video/33225589?key=decaa65dd7bfb98471149208ad42a340).

The issue arises because `mergeSimilarLists` is triggered, leading to a call to `compareListTypes` (see [source code](https://github.com/odoo/odoo/blob/master/addons/html_editor/static/src/main/list/utils.js#L41)). At this point, `a.firstChild` is `null`, which causes the error.

Reply:
Sorry for the late reply. Actually you were right about this because the issue got reported to us yesterday now that the odoo servers are using 18. However, I don't think returning `a === b` makes sense. If both `a` and `b` are `null`, I don't think they should be considered as having the same list type since none of them are lists to begin with. I think I would rather return `false`, so I would change the below condition to be `!a || !b || a.tagName !== b.tagName`. Also, could you move this particular change into a separate commit please? That way we can easily backport it in 18.0 where the issue can occur too. Thank you !